### DISCUSSION TITLE
Should I Enabling Schema Awareness on production?

### DISCUSSION THREAD

**Discussion by nghiepdev** [urql-graphql/urql#1478] [Q&A] [answered]
I'm following the [Schema Awareness" Page of the Graphcache documentation](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/#getting-your-schema). And default integrates on bold development and production.
```js
import schema from './schema.json';

const cache = cacheExchange({ schema });
const client = createClient({
  exchanges: [populateExchange({ schema }), cache],
});
```
I am wondering if it should be enabled on production to reduce the bundle size.

**Comment by kitten** [comment, marked as answer]
Well, you won't get the same behaviour or benefits if it's not enabled in production. Specifically all of its features do very specific things, like partial results, that if you do develop for in development, you'd also want in production.

That's why the `@urql/introspection` package has a function to Munich the schema, so that it's small enough to always load and integrate.

**Reply by JoviDeCroock** [reply]
An additional note would be that the main benefits of `schema-awareness` are developer help, however disabling `schema-awareness` could give you small differences when it comes to partial data. https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/#partial-results

The cost of adding a schema shouldn't be too big if you use something like: https://www.graphql-code-generator.com/docs/plugins/urql-introspection
